mailer.rst

```diff
@@ -0,0 +1,653 @@
+Otherwise, it applies the :phpfunction:`strip_tags` PHP function to the original
+HTML contents.
+
+If you want to define the text content yourself, use the ``text()`` method
+explained in the previous sections or the ``textTemplate()`` method provided by
+the ``TemplatedEmail`` class:
+
+.. code-block:: diff
+
+    + use Symfony\Bridge\Twig\Mime\TemplatedEmail;
+
+    $email = (new TemplatedEmail())
+        // ...
+
+        ->htmlTemplate('emails/signup.html.twig')
+    +     ->textTemplate('emails/signup.txt.twig')
+        // ...
+    ;
+
+.. _embedding-images-in-emails-with-twig:
+
+Embedding Images
+~~~~~~~~~~~~~~~~
+
+Instead of dealing with the ``<img src="cid: ..."/>`` syntax explained in the
+previous sections, when using Twig to render email contents you can refer to
+image files as usual. First, to simplify things, define a Twig namespace called
+``images`` that points to whatever directory your images are stored in:
+
+.. code-block:: yaml
+
+    # config/packages/twig.yaml
+    twig:
+        # ...
+
+        paths:
+            # point this anywhere that your images live
+            images: '%kernel.project_dir%/assets/images'
+
+Now, use the special ``email.image()`` Twig helper to embed the images inside
+the email contents:
+
+.. code-block:: html+twig
+
+    {# '@images/' refers to the Twig namespace defined earlier #}
+    <img src="{{ email.image('@images/logo.png') }}" alt="Logo" />
+
+    <h1>Welcome {{ email.toName }}!</h1>
+    {# ... #}
+
+Inlining CSS Styles
+~~~~~~~~~~~~~~~~~~~
+
+Designing the HTML contents of an email is very different from designing a
+normal HTML page. For starters, most email clients only support a subset of all
+CSS features. In addition, popular email clients like Gmail don't support
+defining styles inside ``<style> ... </style>`` sections and you must **inline
+all the CSS styles**.
+
+CSS inlining means that every HTML tag must define a ``style`` attribute with
+all its CSS styles. This can make organizing your CSS a mess. That's why Twig
+provides a ``CssInlinerExtension`` that automates everything for you. Install
+it with:
+
+.. code-block:: terminal
+
+    $ composer require twig/cssinliner-extension
+
+The extension is enabled automatically. To use this, wrap the entire template
+with the ``inline_css`` filter:
+
+.. code-block:: html+twig
+
+    {% filter inline_css %}
+        <style>
+            {# here, define your CSS styles as usual #}
+            h1 {
+                color: #333;
+            }
+        </style>
+
+        <h1>Welcome {{ email.toName }}!</h1>
+        {# ... #}
+    {% endfilter %}
+
+Using External CSS Files
+........................
+
+You can also define CSS styles in external files and pass them as
+arguments to the filter:
+
+.. code-block:: html+twig
+
+    {# '@css/' refers to the Twig namespace defined earlier #}
+    {% filter inline_css(source('@css/email.css')) %}
+        <style>
+            {# here, define your CSS styles as usual #}
```

Comment:
not so sure about this, isn't the idea of this example to show *not* using `<style>` but rather using a CSS file?